### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by loay**
connect to https://github.com/strongloop-internal/scrum-loopback/issues/745

### CODE REVIEW

**Comment by loay** [comment]
@bajtos @superkhau PTAL. Thanks

**Comment by superkhau** [comment]
@loay Lots of small nitpicks, just edited the doc for you instead. LGTM now.

**Comment by superkhau** [comment]
Please squash before merging of course.

File: documentation.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
+# strong-error-handler documentation
+
+This middleware is intended to be used in both development (debug) and production environments.
+
+## Production mode
+
+- Error responses output do not contain sensitive details or  information.
+- Only `safeFields` are generated in the output.
```
Comment (bajtos) [inline documentation.md]: `safeFields` options was not implemented yet, see https://github.com/strongloop/strong-error-handler/issues/7 

```
For 5xx errors, the output contains only the status code and the status name from the HTTP specification.

For 4xx errors, the output contains the full error message (`error.message`) and the contents of the `details` property (`error.details`). The latter is typically used by `ValidationError` to provide machine-readable details about the validation problems.
```

File: documentation.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+## Debug Mode
+
+- Full error stack traces and internal details of any object passed to this module will be sent back to the client in the HTTP responses when an error occurs.
```
Comment (bajtos) [inline documentation.md]:
```
... passed as the error will be ...
```

File: documentation.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+## Response Format
+
+- `strong-error-handler` currently supports HTML, JSON, and plain text responses.
```
Comment (bajtos) [inline documentation.md]: plain text is not supported yet, see https://github.com/strongloop/strong-error-handler/issues/6

**Comment by bajtos** [comment]
Nice.! I left few comments to address, PTAL.

**Comment by bajtos** [comment]
@loay Also please move the documentation into `README.md`, @crandmck can pick it up from there. Try to preserve the existing module description and any other extra information that we already have in README.md.

**Comment by loay** [comment]
@bajtos @superkhau 
I used my commit including Simon's changes and updated README with the differences. PTAL.

**Comment by superkhau** [comment]
Added fixes for more formatting issues, please pull before you rebase/squash. LGTM

**Pull request merged**